cmd/contour/certgen.go

```diff
@@ -44,7 +44,7 @@ func registerCertGen(app *kingpin.Application) (*kingpin.CmdClause, *certgenConf
 	certgenApp.Flag("certificate-lifetime", "Generated certificate lifetime (in days).").Default(strconv.Itoa(certs.DefaultCertificateLifetime)).UintVar(&certgenConfig.Lifetime)
 	certgenApp.Flag("overwrite", "Overwrite existing files or Secrets.").BoolVar(&certgenConfig.Overwrite)
 	certgenApp.Flag("secrets-format", "Specify how to format the generated Kubernetes Secrets.").Default("legacy").StringVar(&certgenConfig.Format)
-	certgenApp.Flag("secrets-name-prefix", "Specify a prefix to be used as the first part of the generated Kubernetes secrets' names.").StringVar(&certgenConfig.NamePrefix)
+	certgenApp.Flag("secrets-name-suffix", "Specify a suffix to be appended to the generated Kubernetes secrets' names.").StringVar(&certgenConfig.NameSuffix)
```

Comment:
should we do some validation this is a valid string that can be included as part of a name? (i think this may be one of the few user inputs that actually end up in a k8s resource name?)

also maybe less important, but do we expect people to start this with a `-` or do we want to insert it for them? (maybe overkill but just occurred to me)

Reply:
> should we do some validation this is a valid string that can be included as part of a name? (i think this may be one of the few user inputs that actually end up in a k8s resource name?)

Probably not a bad idea; namespace should be validated as well, it's not currently. Worth noting that if you output directly to a cluster, e.g. `--kube`, you already get a decent error message from the API server when the create fails:
```
FATA[0000] failed output certificates                    error="failed to write certificates to \"foo\": Secret \"contourcert%!\" is invalid: metadata.name: Invalid value: \"contourcert%!\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')"
```

But if you were outputting to YAML, obviously you wouldn't get this until later when you actually applied it.

> also maybe less important, but do we expect people to start this with a `-` or do we want to insert it for them? (maybe overkill but just occurred to me)

Originally when I implemented the prefix flag I had the `-` automatically added, then thought that maybe some users wouldn't want a `-`, so I switched to just literally appending the provided suffix. I'm still leaning towards not adding it automatically, just because then there's no way to get rid of it if someone doesn't want it, but open to input.

Reply:
Looking at where this could be implemented, I _guess_ it'd go in the `AsSecrets` / `AsLegacySecrets` functions (https://github.com/projectcontour/contour/blob/63342278690d2aacf02b4a1b65c36a5131336de1/internal/certgen/certgen.go#L151-L207), since those are the ones that deal with generating Kubernetes resources, and also are called both by the cmd-line and by the provisioner directly.  Would require adding an error return param. Does that sound reasonable to you @sunjayBhatia?

Reply:
Added validation/tests in https://github.com/projectcontour/contour/pull/4432/commits/eec57cd82b7bf02e9899e4fb4765bfb26d01dba9

Reply:
looks and works great! 👍🏽 

didnt really think about it before but since the suffix in the provisioner path comes from the gateway resource name  it all works great and should have no chance of an invalid string there